src/Chip/Chip.js

```diff
@@ -0,0 +1,211 @@
+      },
+      '&:active': {
+        boxShadow: shadows[1],
+        backgroundColor: emphasize(backgroundColor, 0.12),
+      },
+      cursor: 'pointer',
+    },
+    deletable: {
+      '&:focus': {
+        backgroundColor: emphasize(backgroundColor, 0.08),
+      },
+    },
+    avatar: {
+      marginRight: -4,
+      width: 32,
+      height: 32,
+      fontSize: 16,
+    },
+    avatarChildren: {
+      width: 19,
+      height: 19,
+    },
+    label: {
+      paddingLeft: 12,
+      paddingRight: 12,
+      userSelect: 'none',
+      whiteSpace: 'nowrap',
+      cursor: 'inherit',
+    },
+    deleteIcon: {
+      color: deleteIconColor,
+      '&:hover': {
+        color: fade(deleteIconColor, 0.4),
+      },
+      cursor: 'pointer',
+      margin: '0 4px 0 -8px',
+    },
+  };
+});
+
+/**
+ * Chips represent complex entities in small blocks, such as a contact.
+ *
+ * ```jsx
+ * <Chip avatar={<Avatar>} label="Label text" />
+ * ```
+ */
+export default function Chip(props, context) {
+  const {
+    avatar: avatarProp,
+    className: classNameProp,
+    deleteIconClassName: deleteIconClassNameProp,
+    label,
+    labelClassName: labelClassNameProp,
+    onClick,
+    onKeyDown,
+    onRequestDelete,
+    ...other
+  } = props;
+
+  let chipRef;
+
+  const handleDeleteIconClick = (event) => {
+    // Stop the event from bubbling up to the `Chip`
+    event.stopPropagation();
+    onRequestDelete(event);
+  };
+
+  const handleKeyDown = (event) => {
+    const key = keycode(event);
+
+    if (onClick && (key === 'space' || key === 'enter')) {
+      event.preventDefault();
+      onClick(event);
+    } else if (onRequestDelete && key === 'backspace') {
+      event.preventDefault();
+      onRequestDelete(event);
+    } else if (key === 'esc') {
+      event.preventDefault();
+      chipRef.blur();
+    }
+    onKeyDown(event);
+  };
+
+  const classes = context.styleManager.render(styleSheet);
+  const className = classNames(
+    classes.root,
+    { [classes.clickable]: onClick },
+    { [classes.deletable]: onRequestDelete },
+    classNameProp,
+  );
+  const labelClassName = classNames(classes.label, labelClassNameProp);
+
+  let deleteIconClassName;
+  let deleteIcon = null;
+  if (onRequestDelete) {
+    deleteIconClassName = classNames(classes.deleteIcon, deleteIconClassNameProp);
+    deleteIcon = (<DeleteIcon className={deleteIconClassName} onClick={handleDeleteIconClick} />);
+  }
+
+  let avatar = null;
+  if (avatarProp && isValidElement(avatarProp)) {
+    const avatarClassName = classNames(classes.avatar, avatarProp.props.className);
+    const avatarChildrenClassName = classNames(classes.avatarChildren, avatarProp.props.childrenClassName);
+
+    avatar = cloneElement(avatarProp, {
+      className: avatarClassName,
+      childrenClassName: avatarChildrenClassName,
+    });
+  }
+
+  return (
+    <button
+      className={className}
+      onClick={onClick}
+      tabIndex={0}
```

Comment:
It fixed the issue, but now I'm wondering if there should no `tabIndex` at all when the element don't have `onClick` or `onRequestClose` callbacks?